lib/tower_cli/resources/ad_hoc.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2015, Ansible, Inc.
+# Luke Sneeringer <lsneeringer@ansible.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, unicode_literals
+
+import click
+
+from tower_cli import models, resources
+from tower_cli.api import client
+from tower_cli.utils import debug, exceptions as exc, types
+
+
+class Resource(models.ExeResource):
+    """A resource for ad hoc commands."""
+    cli_help = 'Launch commands based on playbook given at runtime.'
+    endpoint = '/ad_hoc_commands/'
+
+    # Parameters similar to job
+    name = models.Field(unique=True)
+    job_explanation = models.Field(required=False, display=False)
+    created = models.Field(required=False, display=True)
+    status = models.Field(required=False, display=True)
+    elapsed = models.Field(required=False, display=True)
+
+    # Parameters similar to job_template
+    job_type = models.Field(
+        default='run',
+        display=False,
+        show_default=True,
+        type=click.Choice(['run', 'check']),
+    )
+    inventory = models.Field(type=types.Related('inventory'))
+    machine_credential = models.Field(
```

Comment:
Tower-cli has always been this way, but I'm thinking that maybe Tower renamed machine_credential to credential. Both of these appear in the overall credentials list, but I think the reason we have both fields is so that you can make a job template with _both_ a machine credential and cloud credential. Renaming machine_credential to credential sounds like a correct step to take, but I don't know the history well enough and my theory could be incorrect.